src/resolver.rs

```diff
@@ -57,6 +57,8 @@ pub struct VisitorContext<'s> {
     /// Inside the left hand side of an assignment is in the context of the call's POU
     /// `foo(a := a)` actually means: `foo(foo.a := POU.a)`
     call: Option<&'s str>,
+    /// true if visiting a call statement
+    is_call: bool,
```

Comment:
Since we have an optional call field, isn't it possible to  use this to identify if we are in a call?

Reply:
in resolver.rs `visit_call_statement()` 
```
// #604 needed for recursive function calls
let ctx = ctx.set_is_call();
self.visit_statement(&ctx, operator);
let operator_qualifier = self.get_call_name(operator);
let ctx = ctx.with_call(operator_qualifier.as_str());
let parameters = if let Some(parameters) = parameters_stmt {
    self.visit_statement(&ctx, parameters);
    flatten_expression_list(parameters)
}
```
`ctx.with_call(operator_qualifier.as_str());` will set the `call` field but at this point we allready need the annotated function that was called
this annotation happens before at `self.visit_statement(&ctx, operator);`
so we need the info earlier